pkgs/development/python-modules/fugashi/default.nix

```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, buildPythonPackage, cython, mecab, setuptools-scm }:
+
+buildPythonPackage rec {
+  pname = "fugashi";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "polm";
+    repo = pname;
+    rev = "57e4c28d2caa9893778316a1f038df18eb92ffae";
+    sha256 = "sha256-VDqRhJiNDbKFE284EAUS0d5T9cl8kgyHjh+r/HjjDY8=";
+  };
+
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+  propagatedBuildInputs = [ cython ];
+
+  nativeBuildInputs = [ mecab setuptools-scm ];
+
```

Comment:
Would you please add `pytestCheckHook`?

Reply:
🤔 pytest tests are failing:
```
python3.10-fugashi> fugashi/__init__.py:1: in <module>
python3.10-fugashi>     from .fugashi import *
python3.10-fugashi> E   ModuleNotFoundError: No module named 'fugashi.fugashi'
```

Seems like the [compiled cython pacakage](https://github.com/polm/fugashi/blob/master/setup.py#L25-L31) cannot be imported in [\_\_init\_\_.py](https://github.com/polm/fugashi/blob/master/fugashi/__init__.py#L1)

Reply:
I can import this manually, so it seems that the directory where the test runs is the problem.
Please add
```nix
preCheck = ''
  cd fugashi
'';
```
However, this library needs to package [unidic](https://github.com/polm/unidic-py) in order to actually work.

Reply:
Thanks for the tip !
`fugashi` is now successfully imported, though as you said another package is needed.

Tried to package `unidic`, but it did not work.

I also tried to package `ipadic`, and now some tests are passing !
Most other tests result in a :
```
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 0-1: unexpected end of data
```

Reply:
I’m Japanese, so I’m familiar with these kinds of errors.
We need to add `unidic` to `nativeCheckInputs` too.

> Note this will take up 770MB on disk after install. If you want a small package, try [unidic-lite](https://github.com/polm/unidic-lite).
https://github.com/polm/unidic-py#unidic-py


(It might be better to use `unidic-lite` here.)

Reply:
You were right, `unidic-lite` was also needed in `nativeCheckInputs`.
All tests are now green on my end !

Can you verify the package works on your end @natsukium ?
I will then clean/squash the commits.